### PR TITLE
Add Feature - Handle Multicast Requests

### PR DESCRIPTION
**Pull request by NZSmartie**
I'm working on device discovery and saw this wasn't supported in this library. 

When the application layer receives a messaged that was sent via Multicast; It must set `NetPacket_t.MetaInfo.Type = META_INFO_MULTICAST` before handing off to `NetSocket_t`'s `RxCB(...)` 

Example code with NetConn API (LWIP stack)
```c
// struct netbuf* buffer = ...
 if( buffer->toaddr.type == IPADDR_TYPE_V4 )
    {
        packet.Receiver.NetType = IPV4;
        packet.Receiver.NetAddr.IPv4.u32[0] = ip_addr_get_ip4_u32( &buffer->toaddr );
     
        INFO( "Packet sent to %s:%hu", ipaddr_ntoa( &buffer->toaddr ), packet.Receiver.NetPort );
    }
    else if( buffer->addr.type == IPADDR_TYPE_V6 )
    {
        packet.Receiver.NetType = IPV6;
        packet.Receiver.NetAddr.IPv6.u32[0] = ip_2_ip6( &buffer->toaddr )->addr[0];
        packet.Receiver.NetAddr.IPv6.u32[1] = ip_2_ip6( &buffer->toaddr )->addr[1];
        packet.Receiver.NetAddr.IPv6.u32[2] = ip_2_ip6( &buffer->toaddr )->addr[2];
        packet.Receiver.NetAddr.IPv6.u32[3] = ip_2_ip6( &buffer->toaddr )->addr[3];

        INFO( "Packet sent to [%s]:%hu", ipaddr_ntoa( &buffer->toaddr ), packet.Receiver.NetPort );
    }

    if( EpAreEqual( &packet.Receiver, &NetEp_IPv4_mulitcast ) || EpAreEqual( &packet.Receiver, &NetEp_IPv6_mulitcast ) )
        packet.MetaInfo.Type = META_INFO_MULTICAST;
```

### CODE REVIEW

File: coap_main.c [hunk 1]
```diff
@@ -365,17 +375,34 @@ void _rom CoAP_doWork()
...
+			if(Res == HANDLER_OK) {
+				if(pIA->pRespMsg->Code == EMPTY)
+					pIA->pRespMsg->Code = RESP_SUCCESS_CONTENT_2_05; //handler forgot to set code?
 
 			//b) handler has no result and will not deliver	in the future
-			} else if(Res == HANDLER_ERROR && pIA->pRespMsg->Code == EMPTY) {
-				pIA->pRespMsg->Code = RESP_INTERNAL_SERVER_ERROR_5_00; //handler forgot to set code?
+			} else if(Res == HANDLER_ERROR) {
+				if(pIA->pRespMsg->Code == EMPTY)
+					pIA->pRespMsg->Code = RESP_INTERNAL_SERVER_ERROR_5_00; //handler forgot to set code?
```
Comment (Niondir) [inline coap_main.c]: Please add braces to all if-statements

File: coap_main.c [hunk 2]
```diff
@@ -192,7 +192,7 @@ static CoAP_Result_t _rom SendResp(CoAP_Interaction_t* pIA, CoAP_InteractionStat
 
 		pIA->State = nextIAState; //move to next state
 
-		CoAP_EnqueueLastInteraction(pIA); //(re)enqueue interaction for further processing//todo: in die �u�ere statemachine
+		CoAP_EnqueueLastInteraction(pIA); //(re)enqueue interaction for further processing//todo: in die �u�ere statemachine
```
Comment (Niondir) [inline coap_main.c]: No major issue and we should really replace German comments, but it seems there is some encoding issue. Can you just double check if all files are UTF-8 ?

File: coap_main.c [hunk 3]
```diff
@@ -334,17 +334,28 @@ void _rom CoAP_doWork()
...
-				pIA->State = COAP_STATE_RESOURCE_POSTPONE_EMPTY_ACK_SENT;
-
-				CoAP_SetSleepInteraction(pIA, DEFAULT_LEISURE); // Don't respond right away'
-				CoAP_EnqueueLastInteraction(pIA);
-				INFO("Multicast request postponed processing until %d\r\n", pIA->SleepUntil);
-				return;
+			if(pIA->ReqMetaInfo.Type == META_INFO_MULTICAST){
+				// Messages sent via multicast MUST be NON-confirmable.
+				if(pIA->pReqMsg->Type == CON){
+					INFO("Request received from multicast endpoint is not allowed" );
```
Comment (Niondir) [inline coap_main.c]: Please remove the space after the string.

File: coap_main.c (hunk 2 above)
Comment (NZSmartie) [inline coap_main.c]: I'm not sure what happened there. Files appear to be UTF-8, I ended up copying the original text over the mangled. and now appears the same, but Git thinks it's changed 🤷‍

**Pull request merged**